server/controllers/request.js

```diff
@@ -0,0 +1,64 @@
+
+   
+const Requset = require("../models/Request");
+const asyncHandler = require("express-async-handler");
+
+// @route GET /request/
+// @desc Get this user all request 
+// @access Private
+exports.getAllRequest = asyncHandler(async (req, res, next) => {
+    const userId  = req.user.id
+    const requests = await Request.find({userId:userId})
+
+    res.status(200).json({success:{requests}})
+
+})
+
+// @route POST /request/
+// @desc create new request 
+// @access Private
+exports.createRequest = asyncHandler(async (req, res, next) => {
+
+  const {sitterId,start,end,description} = req.body
+  const userId = req.user.id
+
+  const request = await Request.create({userId,sitterId,start,end,description})
+
+  if(request){
+    res.status(201).json({success:{request}})
+  }else{
+    res.status(400)
+    throw new Error('Invalid request data')
+  }
+
+})
+
+// @route PUT /request/:id/edit
+// @desc update request by id
+// @access private
+exports.updateRequestById = asyncHandler(async (req, res, next) => {
+  
+  const {accepted} = req.body
+  
+  const request = await Request.findById(req.params.id)
+
+  if(request){
+    
+    // if accepted is fasle, it means that sitter decline this request
+    if(accepted){
+      request.accepted = accepted
+    }else{
+      request.declined = !accepted
+    }
```

Comment:
This conditional is unusual. Can we make this more intuitive? I mean technically we could just update two properties but I think what would be even better is also taking in a `declined` boolean in the request and then we can use something like `findByIdAndUpdate` instead.

Reply:
That would get rid of this whole conditional! We should bring in middleware validation as well.